### PR TITLE
docs(ranker): add guide on ranker

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [x] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [x] Are your section headers in sentence case?

### CODE REVIEW

File: chapters/ranker.rst [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+To achieve our objectives, Jina has a family of :term:`Executor` named :term:`Ranker`.
+Jina :term:`Ranker` inherits either the :class:`Match2DocRanker` or the :class:`Chunk2DocRanker`.
+For the :class:`Match2DocRanker`, it re-scores the matches with different mechanisms. For the :class:`Chunk2DocRanker`, it aggegrates the chunks to documents.
+
+
+Match2DocRanker
+^^^^^^^^^^^^^^^^
+
+:class:`Match2DocRanker` re-scores the matches for a document. This Ranker is only responsible for calculating new scores and not for the actual sorting. The sorting is handled in the respective :class:`Matches2DocRankDriver`.
```
Comment (nan-wang) [inline chapters/ranker.rst]: Putting the following example here might be useful.

https://github.com/jina-ai/jina/blob/master/jina/drivers/rank/__init__.py#L38

File: chapters/ranker.rst [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+:class:`Chunk2DocRanker` translates the chunk-wise score to the doc-wise score. If you want to aggregate the already existed top-k chunks into top-k  documents, :class:`Chunk2DocRanker` will be your choice.
+
+
+Drivers
+^^^^^^^^^
+Different :term:`Rankers` will need different :term:`Drivers` to be equipped with.
+For :class:`Match2DocRanker`, we need to choose :class:`Matches2DocRankDriver` to get the scores from a :term:`Ranker` and help to resort the scores.
+For :class:`Chunk2DocRanker`, we need to choose :class:`Chunk2DocRankDriver` to extract matches score from chunks and use the executor to compute the rank and assign the resulting matches to the level above.
+
```
Comment (nan-wang) [inline chapters/ranker.rst:50]:
```suggestion
Rankers in Jina hub
^^^^^^^^^^^^^^^
```

File: chapters/ranker.rst [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+^^^^^^^^^^^^^^^
+
+:class:`Chunk2DocRanker` translates the chunk-wise score to the doc-wise score. If you want to aggregate the already existed top-k chunks into top-k  documents, :class:`Chunk2DocRanker` will be your choice.
+
+
+Drivers
+^^^^^^^^^
+Different :term:`Rankers` will need different :term:`Drivers` to be equipped with.
+For :class:`Match2DocRanker`, we need to choose :class:`Matches2DocRankDriver` to get the scores from a :term:`Ranker` and help to resort the scores.
+For :class:`Chunk2DocRanker`, we need to choose :class:`Chunk2DocRankDriver` to extract matches score from chunks and use the executor to compute the rank and assign the resulting matches to the level above.
```
Comment (nan-wang) [inline chapters/ranker.rst]: `AggregateMatches2DocRankDriver` is missing.

**Review by nan-wang — CHANGES_REQUESTED** [review]
Let put the introduction of `Driver` after the `rankers in jina hub`. Or else the logic flow is a bit broken.

File: chapters/ranker.rst (hunk 1 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: Sure! I was thinking about whether add it or not. I will do it tomorrow.

File: chapters/ranker.rst [hunk 4]
```diff
@@ -0,0 +1,155 @@
+==========================================
+A Guide to Jina Ranker
+==========================================
+
+.. meta::
```
Comment (FionnD) [inline chapters/ranker.rst:5]: Love it!

File: chapters/ranker.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+.. note:: This guide assumes you have a basic understanding of Jina, if you haven't, please check out `Jina 101 <https://101.jina.ai>`_ first.
+
+.. contents:: Table of Contents
+    :depth: 2
+
+Motivation
+--------------------
+
+In a searching system, query accuracy is one of the most important aspects that users will concern. In most cases, we expect well-ordered query results to be presented.
```
Comment (FionnD) [inline chapters/ranker.rst]: In a **search system**

Comment (FionnD) [reply chapters/ranker.rst]: **that will concern users.**

File: chapters/ranker.rst [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+For one search request, we may have multiple matches and we want to have the most relevant match or probably top-K matches presented to users. On the other hand, we may also want to aggregate the top-K chunks to top-K matches.
```
Comment (FionnD) [inline chapters/ranker.rst]: remove **probably**

File: chapters/ranker.rst [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+Before you start
+-------------------
+
+We expect you have a clean Python 3.7/3.8/3.9 (virtual) environment.
+Install Jina on your machine:
+
+.. highlight:: bash
+.. code-block:: bash
+
+    pip install -U jina
```
Comment (FionnD) [inline chapters/ranker.rst]: Can you replace with? 

Requirements
-------------------
You have installed the latest stable release of Jina Core according to the instructions found `here
<https://docs.jina.ai/chapters/core/setup/index.html>`_.

File: chapters/ranker.rst (hunk 1 above)
Comment (FionnD) [inline chapters/ranker.rst]: **D**ocument

Comment (FionnD) [reply chapters/ranker.rst]: Check the whole document for this issue.

File: chapters/ranker.rst [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+:class:`Chunk2DocRanker` translates the chunk-wise score to the doc-wise score. If you want to aggregate the already existed top-k chunks into top-k  documents, :class:`Chunk2DocRanker` will be your choice.
+
+
+Drivers
+^^^^^^^^^
+Different :term:`Rankers` will need different :term:`Drivers` to be equipped with.
+For :class:`Match2DocRanker`, we need to choose :class:`Matches2DocRankDriver` to get the scores from a :term:`Ranker` and help to resort the scores.
+For :class:`Chunk2DocRanker`, we need to choose :class:`Chunk2DocRankDriver` to extract matches score from chunks and use the executor to compute the rank and assign the resulting matches to the level above.
+
+Currently, Jina has 3 Rankers in Jina hub.
```
Comment (FionnD) [inline chapters/ranker.rst]: three. 

My rule is any number less than ten should be written in text, for numbers above ten you can use numbers. 

One, 
Eight, 
10201

File: chapters/ranker.rst [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+        ControlRequest:
+          - !ControlReqDriver {}
+        SearchRequest:
+          - !Chunk2DocRankDriver {}
+
+
+Conclusion
+-----------------
+
+In this guide, we introduce why we need and how to use :term:`Ranker`. Apart from that, we provide some concrete examples of how to use :term:`Ranker`.
```
Comment (FionnD) [inline chapters/ranker.rst]: **introduced**

**Review by FionnD — CHANGES_REQUESTED** [review]
Great guide, I learned a lot!

File: chapters/ranker.rst (hunk 8 above)
Comment (CatStark) [reply chapters/ranker.rst]: Why is that?

File: chapters/ranker.rst [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Description
+     - Drivers
+     - Documentation
+   * - :class:`BiMatchRanker`
+     - :class:`Chunk2DocRanker`
+     - Computes document relevance givena a query taking into account the best chunk-hit from both query and doc perspective.
```
Comment (CatStark) [inline chapters/ranker.rst]: Small typo "given a query".
And maybe punctuation? just not to have a long sentence.

File: chapters/ranker.rst [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+Conventional local usage with uses argument (YAML configuration)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. highlight:: bash
+.. code-block:: bash
+
+        jina pod --uses SimpleAggregateRanker.yml
+
+``SimpleAggregateRanker.yml`` can be configured as following:
```
Comment (CatStark) [inline chapters/ranker.rst]: "Can be configured as follows:"

**Review by CatStark — COMMENTED** [review]
Really cool post! just small grammar things

File: chapters/ranker.rst (hunk 8 above)
Comment (FionnD) [reply chapters/ranker.rst]: "In scientific and technical writing, the prevailing style is to write out numbers under ten. "[[Reference](https://www.grammarly.com/blog/when-to-spell-out-numbers/?gclid=Cj0KCQiAj9iBBhCJARIsAE9qRtAeV-59uHLIKyGCmSKOvAfGt-1ckY6UCtq4VfZcvI0RzFfujeg9xcQaAgTDEALw_wcB&gclsrc=aw.ds)] 

Don't know **why**, it's just the style :)

Comment (Yongxuanzhang) [reply chapters/ranker.rst]: Sure! I will keep that in mind. 👍

File: chapters/ranker.rst (hunk 9 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: 👍

File: chapters/ranker.rst (hunk 1 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: ✅

File: chapters/ranker.rst (hunk 7 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: 👍

File: chapters/ranker.rst (hunk 6 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: 👍

File: chapters/ranker.rst (hunk 5 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: 👍

File: chapters/ranker.rst (hunk 10 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: 👍

File: chapters/ranker.rst (hunk 11 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: 👍

File: chapters/ranker.rst (hunk 3 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst]: ✅

File: chapters/ranker.rst (hunk 2 above)
Comment (Yongxuanzhang) [reply chapters/ranker.rst:50]: 👍

**Comment by Yongxuanzhang** [comment]
> Really cool post! just small grammar things

Thanks for your comments! I added some more content. Hope you could help me review it as well. 😄 
https://deploy-preview-132--jina-docs.netlify.app/chapters/ranker.html

**Comment by Yongxuanzhang** [comment]
> Great guide, I learned a lot!

Thanks for your comments! I added some more content. Hope you could help me review it as well. 😄 
https://deploy-preview-132--jina-docs.netlify.app/chapters/ranker.html

**Review by FionnD — APPROVED** [review]

**Review by CatStark — APPROVED** [review]
Very good doc!

**Review by nan-wang — APPROVED** [review]

**Pull request merged**